authority/provisioner/sign_options.go

```diff
@@ -195,6 +196,50 @@ func (v emailAddressesValidator) Valid(req *x509.CertificateRequest) error {
 	return nil
 }
 
+// urisValidator validates the URI SANs of a certificate request.
+type urisValidator []*url.URL
+
+// Valid checks that certificate request IP Addresses match those configured in
+// the bootstrap (token) flow.
+func (v urisValidator) Valid(req *x509.CertificateRequest) error {
+	want := make(map[string]bool)
+	for _, u := range v {
+		want[u.String()] = true
+	}
+	got := make(map[string]bool)
+	for _, u := range req.URIs {
+		got[u.String()] = true
+	}
+	if !reflect.DeepEqual(want, got) {
+		return errors.Errorf("URIs claim failed - got %v, want %v", req.URIs, v)
+	}
+	return nil
+}
+
+func sansValidators(sans []string) []SignOption {
+	dnsNames, ips, emails, uris := x509util.SplitSANs(sans)
+	return []SignOption{dnsNamesValidator(dnsNames), emailAddressesValidator(emails),
+		ipAddressesValidator(ips), urisValidator(uris)}
```

Comment:
As you're formatting this, can you format it like
```go
return []SignOption{
        dnsNamesValidator(dnsNames), 
        emailAddressesValidator(emails),
        ipAddressesValidator(ips), 
        urisValidator(uris),
}
```

Reply:
We might want to convert this to an actual Validator that checks all of them, so we can just add the validator in the `[]SignOption{}` array, see my previous comment int jwk.go